packages/client/src/__helpers__/describeAuthenticatedScenario.ts

```diff
@@ -0,0 +1,26 @@
+import { never } from '@lens-protocol/shared-kernel';
+
+import { setupAuthenticatedScenario, SetupOptions, TestSetup } from './setupAuthenticatedScenario';
+
+function buildDescribeName(options?: SetupOptions): string {
+  if (!options) {
+    return 'and the instance is authenticated with a random wallet';
+  }
+  if (options.withNewProfile && options.withDispatcher) {
+    return 'and the instance is authenticated with a random wallet with a profile and dispatcher';
+  }
+  if (options.withNewProfile) {
+    return 'and the instance is authenticated with a random wallet with a profile';
+  }
+
+  never('withDispatcher cannot be used without withNewProfile');
+}
+
+type GetTestSetupFn = () => TestSetup;
+
+export const describeAuthenticatedScenario =
+  (options?: SetupOptions) => (callback: (f: GetTestSetupFn) => void) => {
+    const getTestSetup = setupAuthenticatedScenario(options);
+
+    describe(buildDescribeName(options), () => callback(getTestSetup));
```

Comment:
Do we need to pass a function at all? Can we give the object with all the testing mocks and the consumer can just access what they need?

Reply:
the thing is that `describe` function of jest is sync, but `beforeAll` contains async calls (to create profile etc), so we only know that `beforeAll` is finished inside the `it` function.
I tried to call getTestSetup right inside `describeAuthenticatedScenario` and that's what it returns. Notice `accessToken`, `profileId`, even `walletAddress` are undefined - as they all are the result of async code.

```
getTestSetup {
      authentication: Authentication {
        api: AuthenticationApi { sdk: [Object] },
        storage: CredentialsStorage {
          refreshTokenStorage: [Storage],
          accessToken: undefined
        }
      },
      profileId: undefined,
      wallet: Wallet {
        _isSigner: true,
        _signingKey: [Function (anonymous)],
        address: '0xe449202461fe4E32d6fEd8D99D240eD1671eA574',
        _mnemonic: [Function (anonymous)],
        provider: null
      },
      walletAddress: undefined
    }
```

Reply:
I pushed the code that you requested here, but it is not working as expected https://github.com/lens-protocol/lens-sdk/commit/e2bf7cc8c6d8ddea72970cdeafb11a7681043715